write_test.go

```diff
@@ -500,43 +565,44 @@ func TestWrite(t *testing.T) {
 				t.Fatalf("Unexpected error when creating tempfile: %v", err)
 			}
 			if err = Write(file, tc.dataset, tc.opts...); err != tc.expectedError {
-				t.Errorf("Write(%v): unexpected error. got: %v, want: %v", tc.dataset, err, tc.expectedError)
+				t.Fatalf("Write(%v): unexpected error. got: %v, want: %v", tc.dataset, err, tc.expectedError)
 			}
 			file.Close()
 
+			if tc.expectedError != nil {
```

Comment:
I think it's okay to reduce nesting by returning early in this case, but it's worth a comment for future readers to drive home why this is happening, perhaps

```suggestion
			// If we expect an error, we do not need to continue to check the value of the written data, so we continue to the next test case.
			if tc.expectedError != nil {
```